### PR TITLE
pybind/mgr: clean up module vs ceph options naming, helpers

### PR DESCRIPTION
**Pull request by liewegas**
High level goal:
- "module options" are the "mgr/$module/foo" options that are scoped to mgr modules
- "ceph options" are the common/options.cc compiled-in options.

This PR changes:
- rename OPTIONS -> MODULE_OPTIONS
- rename get_config -> get_module_option, set_config -> set_module_option
- rename get_option -> get_ceph_option
- clean up some of the intenral names for mgr_module.py <-> BaseMgr[Standby]Module
- add get_ceph_option to the standby module parent

### CODE REVIEW

**Comment by liewegas** [comment]
@LenzGr there was a bunch of stuff in the dashboard test directory with a mock_ prefix that I wasn't sure about... it might also need to get renamed?  (is the mock_ a magic prefix of some sort?)

**Comment by tserong** [comment]
Looks like it's broken the dashboard testing somehow.

File: src/mgr/BaseMgrModule.cc [hunk 1]
```diff
@@ -883,15 +883,15 @@ PyMethodDef BaseMgrModule_methods[] = {
...
-   "Get a configuration value"},
+  {"_ceph_get_module_option", (PyCFunction)ceph_get_module_option, METH_VARARGS,
+   "Get a module configuration option value"},
 
   {"_ceph_get_store_prefix", (PyCFunction)ceph_store_get_prefix, METH_VARARGS,
    "Get all KV store values with a given prefix"},
 
-  {"_ceph_set_config", (PyCFunction)ceph_config_set, METH_VARARGS,
+  {"_ceph_set_module_option", (PyCFunction)ceph_set_module_option, METH_VARARGS,
    "Set a configuration value"},
```
Comment (tserong) [inline src/mgr/BaseMgrModule.cc]:
```suggestion
    "Set a module configuration option value"},
```

File: src/mgr/BaseMgrStandbyModule.cc [hunk 2]
```diff
@@ -133,9 +153,12 @@ PyMethodDef BaseMgrStandbyModule_methods[] = {
   {"_ceph_get_mgr_id", (PyCFunction)ceph_get_mgr_id, METH_NOARGS,
    "Get the name of the Mgr daemon where we are running"},
 
-  {"_ceph_get_config", (PyCFunction)ceph_config_get, METH_VARARGS,
+  {"_ceph_get_module_option", (PyCFunction)ceph_get_module_option, METH_VARARGS,
    "Get a configuration value"},
```
Comment (tserong) [inline src/mgr/BaseMgrStandbyModule.cc]:
```suggestion
    "Get a module configuration option value"},
```

File: src/mgr/BaseMgrStandbyModule.cc [hunk 3]
```diff
@@ -61,21 +61,41 @@ ceph_get_mgr_id(BaseMgrStandbyModule *self, PyObject *args)
...
     return PyString_FromString(value.c_str());
   } else {
-    dout(4) << "ceph_config_get " << what << " not found " << dendl;
+    dout(4) << __func__ << " " << what << " not found " << dendl;
+    Py_RETURN_NONE;
+  }
+}
+
+static PyObject*
+ceph_option_get(BaseMgrStandbyModule *self, PyObject *args)
```
Comment (tserong) [inline src/mgr/BaseMgrStandbyModule.cc]: Aside from the type of the self parameter, this looks like a duplicate of `ceph_option_get()` from BaseMgrModule.cc. It'd be kinda nice to unify the two, but given there's a bunch of other functions already with subtly different implementations between the two classes, that looks like a separate larger piece of potential cleanup work for another time :-)

File: src/pybind/mgr/dashboard/module.py [hunk 4]
```diff
@@ -250,7 +250,7 @@ class Module(MgrModule, CherryPyConfig):
         {'name': 'crt_file'},
         {'name': 'ssl'}
     ]
-    OPTIONS.extend(options_schema_list())
+    MODULE_OPTIONS.extend(options_schema_list())
```
Comment (tserong) [inline src/pybind/mgr/dashboard/module.py]: Should `options_schema_list()` also be renamed to `module_options_schema_list()`?

File: ceph.spec.in [hunk 5]
```diff
@@ -189,6 +189,8 @@ BuildRequires:	which
 BuildRequires:	xfsprogs
 BuildRequires:	xfsprogs-devel
 BuildRequires:	xmlstarlet
+# for dashboard
+BuildRequires:  xmlsec1
```
Comment (liewegas) [inline ceph.spec.in]: is this right, or do we have some separate list of 'make check' dependencies?

Comment (liewegas) [reply ceph.spec.in]: @tserong

File: src/mgr/BaseMgrStandbyModule.cc (hunk 3 above)
Comment (liewegas) [reply src/mgr/BaseMgrStandbyModule.cc]: agree.. i'm going to leave that cleanup for later, though, as this pr is blocking some other stuff

File: src/pybind/mgr/dashboard/module.py (hunk 4 above)
Comment (liewegas) [reply src/pybind/mgr/dashboard/module.py]: it's not part of the mgr/module interface so i'm inclined to leave it for the dashboard folks?

**Comment by liewegas** [comment]
@tserong rebased to fix trivial conflict with deepsea.  ok to merge?  @rjfd ?

**Comment by sebastian-philipp** [comment]
looks sane to me

File: src/pybind/mgr/dashboard/module.py (hunk 4 above)
Comment (tserong) [reply src/pybind/mgr/dashboard/module.py]: Fine by me

File: ceph.spec.in (hunk 5 above)
Comment (tserong) [reply ceph.spec.in]: If it's only used for `make check`, there's a "distro-conditional make check dependencies" section starting on line 271, so it should go in there. Looking at `install-deps.sh` as updated in #24489, we need `BuildRequires: xmlsec1` for Fedora/RHEL and `BuildRequires: libxmlsec1-1` for SUSE.

**Comment by tserong** [comment]
Aside from shuffling that BuildRequires around in the specfile, yep, all good.

File: ceph.spec.in (hunk 5 above)
Comment (liewegas) [reply ceph.spec.in]: updated, can you take a look?

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
hrm, not sure why teh doc build check is getting stuck?  @alfredodeza any ideas?

**Comment by alfredodeza** [comment]
jenkins test docs

**Pull request merged**